sdk/core/azure-core/src/http/curl/curl_session_private.hpp

```diff
@@ -45,6 +45,19 @@ namespace Azure { namespace Core { namespace Http {
     friend class Azure::Core::Test::CurlConnectionPool_connectionPoolTest_Test;
 #endif
   private:
+    /**
+     * @brief Read one expected byte and throw if it is different than the \p expected
+     *
+     */
+    void ReadExpected(Context const& context, uint8_t expected);
+
+    /**
+     * @brief Read `\r\n` from internal buffer or from the wire.
+     *
+     * @remark throw if `\r\n` are not the next data read.
+     */
+    void ReadCRLF(Context const& context);
+
```

Comment:
I will ask the questions, which you can probably expect now:
1) What's the issue exactly that this is addressing and how did we discover this as an issue?
2) How can we add testing, either by unit tests or e2e?

Reply:
Yes, on it.

I should have created this PR as draft :)  .  You were so fast my friend! 👍 

I am working on the test cases and I will add more info about the issue

Reply:
Feel free to turn it to draft while its WIP:
![image](https://user-images.githubusercontent.com/6527137/105080125-3eee9b00-5a45-11eb-9371-16038c9eaee0.png)